getting-started/bookshelf/main_test.go

```diff
@@ -16,47 +16,57 @@ package main
 
 import (
 	"bytes"
+	"context"
 	"fmt"
+	"io/ioutil"
+	"log"
 	"mime/multipart"
 	"net/http/httptest"
 	"os"
 	"strings"
 	"testing"
 
-	"github.com/GoogleCloudPlatform/golang-samples/getting-started/bookshelf"
-	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
-	"github.com/GoogleCloudPlatform/golang-samples/internal/webtest"
+	"github.com/GoogleCloudPlatform/golang-samples/getting-started/bookshelf/internal/webtest"
 )
 
-var wt *webtest.W
+var (
+	wt *webtest.W
+	b  *Bookshelf
+)
 
 func TestMain(m *testing.M) {
+	projectID := os.Getenv("GOLANG_SAMPLES_FIRESTORE_PROJECT")
+	if projectID == "" {
+		fmt.Fprintln(os.Stderr, "GOLANG_SAMPLES_FIRESTORE_PROJECT not set. Skipping.")
```

Comment:
use in-memory DB for tests, firestore for e2e tests?

Reply:
Done. It ended up changing the API quite a bit -- now pass a `BookDatabase` to `NewBookshelf`. It makes sense for testing, but is a slightly awkward API.

I updated this to always test the in memory database and only test the Firestore one if it's able to.